### PR TITLE
Convert shapenet into kubric assets

### LINKED ISSUE #33: ShapeNet preprocessing
Script for preprocessing ShapeNet so that it can be used as an AssetSource

- [x] prepare a docker container that contains all dependencies for the task
- [x] prepare a Makefile to replicate all efforts in standalone

### PR DESCRIPTION
**Pull request by taiya**

### CODE REVIEW

File: shapenet2kubric/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+RUN apt-get update --fix-missing
+RUN apt-get -y install cmake
+
+# --- install obj2gltf (REQUIRED)
+RUN apt install -y nodejs
+RUN apt install -y npm
+RUN npm install -g npm@6.9.0
+RUN npm install -g obj2gltf
+
+# --- Blender2.83 uses the system python3.7
```
Comment (Qwlouse) [inline shapenet2kubric/Dockerfile:13]: We don't need blender here at all do we?

**Comment by Qwlouse** [comment]
Not sure we are allowed to include bathtub.obj because this is part of shapenet isn't it?

The script superficially looks good to me. But I haven't yet tested / verified that it runs correctly yet. Do you have an example of a converted asset?

**Comment by taiya** [comment]
@Qwlouse added prompt about licence, please check/test?

File: shapenet2kubric/Dockerfile (hunk 1 above)
Comment (taiya) [reply shapenet2kubric/Dockerfile:13]: Yes, it's gone now, I only keep python3.7 in the latest commit.
Is that the best solution for it though?

**Comment by IssamLaradji** [comment]
I tested `make container_pull` and `make convert` and it works great! it might be a good idea to allow the user to define the dataset path.

On another note, the bathtub OBJ is not from ShapeNet, it is part of ManifoldPlus repo, https://github.com/hjwdzh/ManifoldPlus/tree/master/data, do you think we can `wget` it directly from there?

**Comment by Qwlouse** [comment]
`make obj2gltf_test` complains because `bathtub_input.mtl` is missing

```
$ make obj2gltf_test
docker run --rm --interactive \
		--user 584848:89939 \
		--volume "/Users/klausg/kubric/kubric/shapenet2kubric:/shapenet2kubric" \
		kubricdockerhub/shapenet:latest \
		obj2gltf \
			-i /shapenet2kubric/bathtub.obj \
			-o /shapenet2kubric/bathtub
ENOENT: no such file or directory, open '/shapenet2kubric/bathtub_input.mtl'
Could not read material file at /shapenet2kubric/bathtub_input.mtl. Attempting to read the material file from within the obj directory instead.
ENOENT: no such file or directory, open '/shapenet2kubric/bathtub_input.mtl'
Could not read material file at /shapenet2kubric/bathtub_input.mtl. Using default material instead.
Total: 803.809ms
```

**Comment by Qwlouse** [comment]
`make convert` fails because of the `.DS_Store` file that Mac automatically creates. 
Could be solved by only collecting directories [here](https://github.com/google-research/kubric/pull/112/files#diff-b5d7ab7a0554d44f1fe06bd9627e5164c0ddca149f57b68aa1f9a067575235ddR272)

```
$  make convert
docker run --rm --interactive \
		--user 584848:89939 \
		--volume "/Users/klausg/kubric/kubric/shapenet2kubric:/shapenet2kubric" \
		--volume "/Users/klausg/datasets/ShapeNetCore.v2:/ShapeNetCore.v2" \
		--volume "/Users/klausg/datasets/ShapeNetCore.v2.kubric:/ShapeNetCore.v2.kubric" \
		kubricdockerhub/shapenet:latest \
		python3.7 shapenet2kubric/convert.py \
			--datadir "/ShapeNetCore.v2" \
			--outdir "/ShapeNetCore.v2.kubric"
pybullet build time: Jun  8 2021 23:25:51
Traceback (most recent call last):
  File "shapenet2kubric/convert.py", line 276, in <module>
    obj_id_list = os.listdir(os.path.join(args.datadir, cat_dir))
NotADirectoryError: [Errno 20] Not a directory: '/ShapeNetCore.v2/.DS_Store'
make: *** [convert] Error 1
```

**Comment by Qwlouse** [comment]
Other than the above the script seems to run on my machine. But I estimate that it would take about three weeks on my laptop, so we should add some logic for processing assets in parallel.

**Comment by taiya** [comment]
> it might be a good idea to allow the user to define the dataset path.

Done, you can now `make download DOWNLOADPREFIX=/path/you/want`

> On another note, the bathtub OBJ is not from ShapeNet, it is part of ManifoldPlus repo, https://github.com/hjwdzh/ManifoldPlus/tree/master/data, do you think we can `wget` it directly from there?

Done → see latest commit

> `make convert` fails because of the `.DS_Store` file that Mac automatically creates.
> Could be solved by only collecting directories [here](https://github.com/google-research/kubric/pull/112/files#diff-b5d7ab7a0554d44f1fe06bd9627e5164c0ddca149f57b68aa1f9a067575235ddR272)

Done (also using pathlib vs. os now).

**Comment by taiya** [comment]
Alright, after some frustrating time with `obj2gltf` (not sure why my local docker does not provide identical execution to my cloudtop docker), I have a first positive outcome. The converted model renders correctly in both a GLTF2 viewer, as well as renders correctly in blender:

Screen Shot 2021-07-28 at 6 15 32 PM

NOTE: one point of pain is that it loads as a large collection of sub-meshes in blender (not sure if you took care of this already @Qwlouse?)

**Comment by taiya** [comment]
I sang too early, this doesn't look that good:
Screen Shot 2021-07-28 at 10 51 05 PM

**Comment by Qwlouse** [comment]
Current run looks very promising. I'm merging and closing.

**Pull request merged**